hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/HapiTestEngine.java

```diff
@@ -243,33 +250,67 @@ public SkipResult shouldBeSkipped(HapiTestEngineExecutionContext context) {
      */
     private static final class MethodTestDescriptor extends AbstractTestDescriptor
             implements Node<HapiTestEngineExecutionContext> {
+
         /** The method under test */
         private final Method testMethod;
 
+        private final Set<TestTag> testTags;
+
         public MethodTestDescriptor(Method testMethod, ClassTestDescriptor parent) {
             super(
                     parent.getUniqueId().append("method", testMethod.getName()),
                     testMethod.getName(),
                     MethodSource.from(testMethod));
             this.testMethod = testMethod;
+            this.testTags = getTagsIfAny(testMethod);
             setParent(parent);
         }
 
+        private Set<TestTag> getTagsIfAny(Method testMethod) {
+            // When a method has a single @Tag annotation, we retrieve it by filtering for Tag.class.
+            // In cases where a method has multiple @Tag annotations, we use Tags.class to access all of them.
+            // Ideally, Tags.class should encompass both single and multiple @Tag annotations,
+            // but the current implementation does not support this.
+            final var tagsAnnotation = testMethod.getAnnotation(Tags.class);
+            final var tagAnnotation = testMethod.getAnnotation(Tag.class);
+
+            final var tags = new HashSet<TestTag>();
+            if (tagsAnnotation != null) {
+                tags.addAll(Arrays.stream(tagsAnnotation.value())
+                        .map(t -> TestTag.create(t.value()))
+                        .toList());
+            } else if (tagAnnotation != null) {
```

Comment:
Does it make sense to support both annotations on a method?

Reply:
I think it makes sense, yes. For example we might have a test that we can annotate with **slow** and **performance** tags at the same time